test/crit/main.go

```diff
@@ -189,3 +190,73 @@ func readMemoryPages() error {
 
 	return nil
 }
+
+// testGetShmemSize tests the GetShmemSize method of the MemoryReader struct.
+func testGetShmemSize() error {
+	testCases := []struct {
+		imgsDir           string
+		expectedShmemSize int64
+	}{
+		{
+			imgsDir:           loopTestImgDir,
+			expectedShmemSize: 0,
+		},
+		{
+			imgsDir:           private4kmmTestImgDir,
+			expectedShmemSize: 0,
+		},
+		{
+			imgsDir:           shared4kmmTestImgDir,
+			expectedShmemSize: 4096,
+		},
+		{
+			imgsDir:           shared32kmmTestImgDir,
+			expectedShmemSize: 32768,
```

Comment:
Would it be possible to add also tests that are not multiple of 2 (e.g, 1025, or 4196)?